lib/server_connection.ml

```diff
@@ -221,9 +221,18 @@ let rec _next_read_operation t =
     Reader.next t.reader
   ) else (
     let reqd = current_reqd_exn t in
+    (* XXX(dpatti): This fails for the same reason as my comment below in the
+       final_read_operation section. I played around with some alternatives and
+       believe I have some more improvements to the request queue mechanism that
+       removes the need for two hacks. *)
```

Comment:
Boy I wish I remember what I meant to do here. I'll think about it a bit and update this comment, I don't think this needs to necessarily block anything.

Reply:
Cleaned this thread of conversation up in the code.